components/brave_wallet_ui/page/screens/fund-wallet/android/deposit-funds.tsx

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { render } from 'react-dom'
+import { Route, Switch, BrowserRouter, useHistory } from 'react-router-dom'
+import { Provider } from 'react-redux'
+
+import { initLocale } from 'brave-ui'
+
+// style
+import walletDarkTheme from '../../../../theme/wallet-dark'
+import walletLightTheme from '../../../../theme/wallet-light'
+import 'emptykit.css'
+
+// Utils
+import { loadTimeData } from '../../../../../common/loadTimeData'
+import * as Lib from '../../../../common/async/lib'
+import { getLocale } from '../../../../../common/locale'
+
+// actions
+import * as WalletActions from '../../../../common/actions/wallet_actions'
+
+// Components
+import { store, walletPageApiProxy } from '../../../store'
+import BraveCoreThemeProvider
+  from '../../../../../common/BraveCoreThemeProvider'
+import { DepositFundsScreen } from '../deposit-funds'
+import { LibContext } from '../../../../common/context/lib.context'
+import { ApiProxyContext }
+  from '../../../../common/context/api-proxy.context'
+
+import { WalletPageWrapper }
+  from '../../../../components/desktop/wallet-page-wrapper/wallet-page-wrapper'
+import { PageTitleHeader }
+  from '../../../../components/desktop/card-headers/page-title-header'
+
+import { setIconBasePath } from '@brave/leo/react/icon'
+setIconBasePath('chrome://resources/brave-icons')
+
+export function AndroidDepositApp() {
+  let history = useHistory()
+
+  const [showDepositAddress, setShowDepositAddress] = React.useState<boolean>(false)
+
+  const handleDepositScreenBack = React.useCallback(() => {
+    if (!showDepositAddress && history.length) {
```

Comment:
does history ever have a length on android in this case?

Reply:
No, good catch, it can be safely removed.